cs50/2017/x/find/less/check50/__init__.py

```diff
@@ -15,6 +15,27 @@ def compiles(self):
         self.add("find.c")
         self.spawn("clang -std=c11 -o find find.c helpers.c -lcs50 -lm").exit(0)
 
+    def test_sorted(self, items):
+        p = self.spawn("./sort")
+        for i in items:
+            p.stdin(str(i))
+        p.stdin(EOF)
+        for i in sorted(items):
+            p.stdout(str(i))
+        p.exit(0)
+
+    @check("compiles")
+    def sort_reversed(self):
+        """sorts {5,4,3,2,1}"""
+        self.add("sort.c")
+        self.spawn("clang -std=c11 -o sort sort.c helpers.c -lcs50 -lm").exit(0)
+        self.test_sorted([5, 4, 3, 2, 1])
+
+    @check("sort_reversed")
```

Comment:
@tlively, any reason this one's dependent on `sort_reversed`?

Reply:
because `sort_reversed` also compiles sort.c. The reason there isn't mutual dependency that does this instead is that I don't want a student visible test mentioning sort.c. I also don't want to compile sort.c twice, which would be another solution.

Reply:
Interesting scenario. We don't currently have a mechanism for caching intermediate working dirs that aren't themselves checks (as `compiles` is); ergo this workaround. Though I think this workaround could accidentally confuse, since it implies to a student that one of the checks is dependent on getting a reversed array to sort, which isn't necessarily the case, nor a helpful hint/implication. Am inclined to think we should compile `sort.c` in `compiles` up top, even though most checks won't use it?

CC @brianyu28, @crossroads1112